addon/components/variable-plugin/insert-variable-card.ts

```diff
@@ -1,193 +1,38 @@
-        if (variableType) {
-          variablesArray.push(variableType);
-        } else {
-          console.warn(
-            `Template Variable Plugin: variable type ${type} not found in the default variable types`,
-          );
-        }
-      } else {
-        variablesArray.push(type);
-      }
-    }
-    this.variablesArray = variablesArray;
-  }
-
   get controller() {
     return this.args.controller;
   }
 
-  get numberVariableError() {
-    const minVal = this.extraAttributes.minimumValue;
-    const maxVal = this.extraAttributes.maximumValue;
-    if (
-      isNumber(minVal) &&
-      isNumber(maxVal) &&
-      Number(minVal) > Number(maxVal)
-    ) {
-      return this.intl.t('variable.number.error-min-bigger-than-max');
-    }
-
-    return '';
-  }
-
   @action
-  updateVariableLabel(event: InputEvent) {
-    this.variableLabel = (event.target as HTMLInputElement).value;
-  }
-
-  @action
-  insert() {
-    if (!this.selectedVariable || this.numberVariableError) {
-      return;
-    }
-
-    const node = this.selectedVariable.constructor({
-      schema: this.controller.schema,
-      codelist: this.selectedSubtype,
-      label: this.variableLabel !== '' ? this.variableLabel : undefined,
-      attributes: {
-        source: this.endpoint,
-        ...this.extraAttributes.asObject(),
-      },
-    });
-
-    this.variableLabel = '';
-    this.extraAttributes.reset();
-
-    this.controller.withTransaction(
-      (tr) => {
-        return tr.replaceSelectionWith(node);
-      },
-      { view: this.controller.mainEditorView },
-    );
-  }
-
-  @action
-  updateSelectedVariable(variable: VariableType) {
+  updateSelectedVariable(variable: VariableConfig) {
     this.selectedVariable = variable;
-    if (variable.fetchSubtypes) {
-      void this.fetchSubtypes.perform(variable.fetchSubtypes);
-      this.hasSubtype = true;
-    } else {
-      this.hasSubtype = false;
-    }
-  }
-
-  fetchSubtypes = task(
-    async (
-      fetchFunction: (
-        endpoint: string,
-        publisher: string,
-      ) => Promise<CodeList[]>,
-    ) => {
-      const subtypes = await fetchFunction(this.endpoint, this.publisher);
-      this.subtypes = subtypes;
-    },
-  );
-
-  @action
-  updateSubtype(subtype: CodeList) {
-    this.selectedSubtype = subtype;
-    this.extraAttributes.reset();
-  }
-
-  get type() {
-    return this.selectedVariable?.label;
   }
 
   get showCard() {
-    if (this.args.controller.inEmbeddedView) {
-      return false;
-    }
-    const { selection } = this.args.controller.mainEditorState;
-    if (
-      selection instanceof NodeSelection &&
-      selection.node.type === this.args.controller.schema.nodes.variable
-    ) {
-      return false;
-    } else {
-      const variable = findParentNodeOfType(
-        this.args.controller.schema.nodes.variable,
-      )(selection);
-      return !variable;
-    }
+    return !this.controller.inEmbeddedView;
```

Comment:
This makes it so the card will still show when selecting the date or number variable (as they don't use the embedded view).
This is not bad perse: it will just replace the number or date variable when inserting (so a logical functionality).

We can leave it as is, but it is still a bit inconsistent.
Potential solutions: also hide when it is a node selection or even specifically a variable selection (but that's probably harder?)

Reply:
I can do a check if the selected node has group `variable` for example?

Reply:
Yes I honestly think that would be a needed improvement, because then it is based on the structure/content instead of implementation details (using embedded editor or not).

Reply:
Updated it :)